cf/api/domains.go

```diff
@@ -17,14 +17,14 @@ import (
 //go:generate counterfeiter -o fakes/fake_domain_repository.go . DomainRepository
 type DomainRepository interface {
 	ListDomainsForOrg(orgGuid string, cb func(models.DomainFields) bool) error
-	FindSharedByName(name string) (domain models.DomainFields, apiErr error)
-	FindPrivateByName(name string) (domain models.DomainFields, apiErr error)
-	FindByNameInOrg(name string, owningOrgGuid string) (domain models.DomainFields, apiErr error)
-	Create(domainName string, owningOrgGuid string) (createdDomain models.DomainFields, apiErr error)
-	CreateSharedDomain(domainName string) (apiErr error)
-	Delete(domainGuid string) (apiErr error)
-	DeleteSharedDomain(domainGuid string) (apiErr error)
-	FirstOrDefault(orgGuid string, name *string) (domain models.DomainFields, error error)
+	FindSharedByName(name string) (models.DomainFields, error)
+	FindPrivateByName(name string) (models.DomainFields, error)
+	FindByNameInOrg(name string, owningOrgGuid string) (models.DomainFields, error)
+	Create(domainName string, owningOrgGuid string) (models.DomainFields, error)
+	CreateSharedDomain(domainName string, routerGroupGuid string) error
+	Delete(domainGuid string) error
+	DeleteSharedDomain(domainGuid string) error
+	FirstOrDefault(orgGuid string, name *string) (models.DomainFields, error)
 }
```

Comment:
The methods which haven't actually changed in implementation here need to not change at all, including by not removing the apiErr return args. The tests for the code may not support the changes required to safely remove the named return args in the other methods.

Separately, it would be better to have done such changes in a different commit rather than to include them in the commit for the story.